python/mach_bootstrap.py

```diff
@@ -131,45 +111,32 @@ def install_virtual_env_requirements(project_path: str, python: str, virtualenv_
     requirements_hash = requirements_hasher.hexdigest()
 
     if marker_hash != requirements_hash:
-        print(" * Upgrading pip...")
-        _process_exec([python, "-m", "pip", "install", "--upgrade", "pip"])
-
         print(" * Installing Python requirements...")
-        _process_exec([python, "-m", "pip", "install", "-I",
-                       "-r", requirements_paths[0],
-                       "-r", requirements_paths[1],
-                       "-r", requirements_paths[2]])
+        pip_install_command = ["uv", "pip", "install"]
+        for requirements in requirements_paths:
+            pip_install_command.extend(["-r", requirements])
+        _process_exec(pip_install_command, cwd=project_path)
         with open(marker_path, "w") as marker_file:
             marker_file.write(requirements_hash)
 
 
 def _activate_virtualenv(topdir):
-    virtualenv_path = os.path.join(topdir, "python", "_venv%d.%d" % (sys.version_info[0], sys.version_info[1]))
-    python = sys.executable
+    virtualenv_path = os.path.join(topdir, ".venv")
 
     if os.environ.get("VIRTUAL_ENV") != virtualenv_path:
```

Comment:
This is still causing problems when doing `python3 ./mach build`, assuming python3 is e.g. python3.13. 
Just ./mach build works fine, because we re-execute with `uv`, but here we are still using the system python and not switching to the new python.

Reply:
Right, invoking `./mach` via external python is still not supported. That is currently only done in CI where we pin the version ,but may we should switch to `./mach` there as well. I thought checking the python version and aborting for this case where use runs mach using external python, but I wasn't sure if that is worth it as our guides almost always use `./mach` directly.

Reply:
It's not very obvious that running a python script directly in python is not supposed to work.
Could we perhaps make `./mach` just a shell script, so that running it in python directly just won't work? It looks like at this point the script is basically just shell-script anyway.

Reply:
I originally had it as pure shell script that calls `uv run python/mach_bootstrap.py`, but that resulted in failures due to some modules derviving relative paths from `sys.args[0]` and the `mach help` message also deriving the wrong executable name when listing the available commands  probably due to same `sys.arg[0]` issue. 

It might be possible to fix that with some more tricks, but I didn't get time to do that in this PR.